### DISCUSSION TITLE
how to aggregate nested array of objects in single view

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#300] [Q&A] [answered]
### Which section?

Fast Data

### Which is the version?

Mia Platform v11.4.2

### Description

Hi,
How can I add, using the no code aggregation, a field that is an array of object, in which inside the object there is another array of object?
The single view I want to obtain is structured like this this:
```
{
   "foo":[
      {
         "bar":[
            {
               "pippo":"pluto"
            }
         ]
      }
   ]
}
```

I wrote this JSON schema for the first level field:
```
{
  "properties": {
    "name": {
      "type": "string"
    },
    "allergens": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "name": "string",
          "description": "string"
        }
      }
    }
  },
  "type": "object"
}
```
but I see this error in the aggregation no code:
![image](https://github.com/mia-platform/community/assets/18056946/8771c501-a6ae-4dc5-8f3e-0b7e67046266)

How can I obtain the desired result?

**Comment by danibix95** [comment, marked as answer]
I believe your JSON schema is invalid since is missing the type in `name` and `description` fields definition.
It should look something similar to this:

```json
{
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "allergens": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "name": {
            "type": "string"
          },
          "description": {
            "type": "string"
          }
        }
      }
    }
  }
}
```

Once the JSON schema for that property is fixed, the No Code Aggregator should work without any further issue.

**Reply by elmalakomar** [reply]
omg, you are correct.
Thanks a lot @danibix95
